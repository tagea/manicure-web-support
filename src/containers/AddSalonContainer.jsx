import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { AddSalon } from '../components/AddSalon';

export class AddSalonContainer extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {};
  }

  changeInput = (ev) => {
    this.setState({
      [ev.target.name]: ev.target.value
    });
  }

  render() {
    const { ...values } = this.state;
    return (
      <AddSalon changeInput={this.changeInput} values={values} />
    );
  }
}

AddSalonContainer.propTypes = {
};

AddSalonContainer.defaultProps = {
};

const mapStateToProps = state => ({
});

const mapDispatchToProps = dispatch => ({
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AddSalonContainer);
