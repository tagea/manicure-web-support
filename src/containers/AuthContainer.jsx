import React, { PureComponent } from 'react';
import { Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { Auth } from '../components/Auth';
import { signInAction, stayInAction } from '../services/auth/actions';

import { ROUTE_PATHS } from '../routes';

export class AuthContainer extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      username: '',
      password: ''
    };
  }

  componentDidMount() {
    const { stayIn } = this.props;
    stayIn();
  }

  changeInput = (ev) => {
    this.setState({
      [ev.target.name]: ev.target.value
    });
  }

  pressEnter = (ev) => {
    if (ev.key === 'Enter') {
      this.handleSubmit();
    }
  }

  handleSubmit = () => {
    const { username, password } = this.state;
    const { signIn } = this.props;
    signIn(username, password);
  }

  render() {
    const { error, authenticated } = this.props;
    if (authenticated) {
      return <Redirect to={ROUTE_PATHS.root} />;
    }
    return (
      <Auth
        changeInput={this.changeInput}
        pressEnter={this.pressEnter}
        handleSubmit={this.handleSubmit}
        error={error}
      />
    );
  }
}

AuthContainer.propTypes = {
  error: PropTypes.bool,
  authenticated: PropTypes.bool,
  stayIn: PropTypes.func.isRequired,
  signIn: PropTypes.func.isRequired
};

AuthContainer.defaultProps = {
  error: false,
  authenticated: false
};


const mapStateToProps = state => ({
  authenticated: state.auth.authenticated,
  error: state.auth.error
});

const mapDispatchToProps = dispatch => ({
  stayIn: () => dispatch(stayInAction()),
  signIn: (username, password) => dispatch(signInAction(username, password))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AuthContainer);
