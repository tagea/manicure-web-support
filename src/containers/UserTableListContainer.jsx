import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { TableList } from '../components/TableList';
import { fetchUserAction } from '../services/users/actions';
import { headerUser } from '../utils/headersList';

import { ROUTE_PATHS } from '../routes';

class UserTableListContainer extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      data: [],
      querySize: 10,
      totalSize: 0,
      pageNumber: 1,
      waiting: false
    };
  }

  componentDidMount() {
    this.loadMore();
  }

  componentWillUnmount() {
    this.actuallyLoadMore = () => false;
  }

  // load part of data
  loadMore = () => {
    const { data, querySize, pageNumber } = this.state;
    const { fetchData, nextPage } = this.props;
    if (data.length === 0 || nextPage !== null) {
      fetchData(pageNumber, querySize)
        .then(() => {
          this.actuallyLoadMore();
        });
    }
  };

  actuallyLoadMore = () => {
    const { data, pageNumber } = this.state;
    const { lastLoadedData, totalSize, waiting } = this.props;
    this.setState({
      data: data.concat(lastLoadedData),
      pageNumber: pageNumber + 1,
      totalSize,
      waiting
    });
  }

  render() {
    const { data, totalSize, waiting } = this.state;
    const { error } = this.props;
    if (error !== null) {
      return <Redirect to={ROUTE_PATHS.auth} />;
    }
    return (
      <TableList
        waiting={waiting}
        loadMoreRows={this.loadMore}
        listData={data}
        columns={headerUser}
        rowCount={totalSize}
      />
    );
  }
}

UserTableListContainer.propTypes = {
  fetchData: PropTypes.func.isRequired,
  waiting: PropTypes.bool,
  lastLoadedData: PropTypes.instanceOf(Array),
  totalSize: PropTypes.number,
  nextPage: PropTypes.string,
  error: PropTypes.string
};

UserTableListContainer.defaultProps = {
  waiting: false,
  lastLoadedData: [],
  totalSize: 0,
  nextPage: null,
  error: null
};

const mapStateToProps = state => ({
  waiting: state.users.waiting,
  lastLoadedData: state.users.users,
  totalSize: state.users.sizeOfUsers,
  nextPage: state.users.nextPageUsers,
  error: state.users.error
});

const mapDispatchToProps = dispatch => ({
  fetchData: (pageNumber, size) => dispatch(fetchUserAction(pageNumber, size))
});


export default connect(
  mapStateToProps,
  mapDispatchToProps
)(UserTableListContainer);
