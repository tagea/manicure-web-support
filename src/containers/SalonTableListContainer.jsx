import React, { PureComponent, Fragment } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { TableList } from '../components/TableList';
import { ModalWindow } from '../components/ModalWindow';
import { AddSalonContainer } from './AddSalonContainer';
import { fetchSalonAction } from '../services/salons/actions';
import { headerSalon } from '../utils/headersList';

class SalonTableListContainer extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      data: [],
      querySize: 10,
      totalSize: 0,
      pageNumber: 1,
      waiting: false,
      openModal: false,
      titleWindow: ''
    };
  }

  componentDidMount() {
    this.loadMore();
  }

  componentWillUnmount() {
    this.actuallyLoadMore = () => false;
  }

  // load part of data
  loadMore = () => {
    const { data, querySize, pageNumber } = this.state;
    const { fetchData, nextPage } = this.props;
    if (data.length === 0 || nextPage !== null) {
      fetchData(pageNumber, querySize)
        .then(() => {
          this.actuallyLoadMore();
        });
    }
  };

  actuallyLoadMore = () => {
    const { data, pageNumber } = this.state;
    const { lastLoadedData, totalSize, waiting } = this.props;
    this.setState({
      data: data.concat(lastLoadedData),
      pageNumber: pageNumber + 1,
      totalSize,
      waiting
    });
  }

  openAddSalonDialog = () => {
    this.setState({
      openModal: true,
      titleWindow: 'Add a new Salon'
    });
  }

  handleClose = () => {
    this.setState({
      openModal: false
    });
  }

  render() {
    const { data, totalSize, waiting, openModal, titleWindow } = this.state;

    return (
      <Fragment>
        <TableList
          waiting={waiting}
          loadMoreRows={this.loadMore}
          listData={data}
          columns={headerSalon}
          handleOpen={this.openAddSalonDialog}
          rowCount={totalSize}
        />
        <ModalWindow
          open={openModal}
          closeModal={this.handleClose}
          titleWindow={titleWindow}
        >
          <AddSalonContainer />
        </ModalWindow>
      </Fragment>
    );
  }
}

SalonTableListContainer.propTypes = {
  waiting: PropTypes.bool,
  fetchData: PropTypes.func.isRequired,
  lastLoadedData: PropTypes.instanceOf(Array),
  totalSize: PropTypes.number,
  nextPage: PropTypes.string
};

SalonTableListContainer.defaultProps = {
  waiting: false,
  lastLoadedData: [],
  totalSize: 0,
  nextPage: null
};

const mapStateToProps = state => ({
  waiting: state.salons.waiting,
  lastLoadedData: state.salons.salons,
  totalSize: state.salons.sizeOfSalons,
  nextPage: state.salons.nextPageSalons
});

const mapDispatchToProps = dispatch => ({
  fetchData: (pageNumber, size) => dispatch(fetchSalonAction(pageNumber, size))
});


export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SalonTableListContainer);
