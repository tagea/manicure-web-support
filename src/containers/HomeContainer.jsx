import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import { Redirect } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Toolbar } from '../components/Toolbar';
import { Home } from '../components/Home';
import { HomeTabs } from '../components/HomeTabs';
import UserTableListContainer from './UserTableListContainer';
import SalonTableListContainer from './SalonTableListContainer';

import { signOutAction, stayInAction } from '../services/auth/actions';

import { ROUTE_PATHS } from '../routes';

class HomeContainer extends PureComponent {
  componentDidMount() {
    const { stayIn } = this.props;
    stayIn();
  }

  handleSubmit = () => {
    const { signOut } = this.props;
    signOut();
  }

  render() {
    const { authenticated } = this.props;
    if (authenticated) {
      return (
        <Home>
          <Toolbar />
          <HomeTabs titles={['Users', 'Salons']} handleSubmit={this.handleSubmit}>
            <UserTableListContainer />
            <SalonTableListContainer />
          </HomeTabs>
        </Home>
      );
    }
    return (
      <Redirect to={ROUTE_PATHS.auth} />
    );
  }
}

HomeContainer.propTypes = {
  authenticated: PropTypes.bool,
  stayIn: PropTypes.func.isRequired,
  signOut: PropTypes.func.isRequired
};

HomeContainer.defaultProps = {
  authenticated: false
};


const mapStateToProps = state => ({
  authenticated: state.auth.authenticated
});

const mapDispatchToProps = dispatch => ({
  stayIn: () => dispatch(stayInAction()),
  signOut: () => dispatch(signOutAction())
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(HomeContainer);
