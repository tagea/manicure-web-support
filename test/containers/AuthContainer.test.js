import React from 'react';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { AuthContainer } from '../../src/containers/AuthContainer';

const initSetup = () => {
  const handleSubmit = jest.fn();
  const props = {
    stayIn: jest.fn(),
    signIn: jest.fn(),
    handleSubmit
  };
  const mockStore = configureStore();
  const store = mockStore({});
  const AuthContainerShallow = shallow(<AuthContainer store={store} {...props} />);
  const AuthContainerMount = mount(<AuthContainer store={store} {...props} />);

  return {
    AuthContainerShallow,
    AuthContainerMount
  };
};

it('render AuthContainer', () => {
  const { AuthContainerShallow } = initSetup();
  expect(AuthContainerShallow).toMatchSnapshot();
});

it('check username in store after onChange', () => {
  const { AuthContainerMount } = initSetup();
  const InputTest = AuthContainerMount.find('input').first();
  InputTest.simulate('change', { target: { name: 'username', value: 'user' } });
  expect(AuthContainerMount.state('username')).toEqual('user');
});

it('check password in store after onChange callback', () => {
  const { AuthContainerMount } = initSetup();
  const InputTest = AuthContainerMount.find('input').last();
  InputTest.simulate('change', { target: { name: 'password', value: '123' } });
  expect(AuthContainerMount.state('password')).toEqual('123');
});

it('check handleSubmit callback', () => {
  const { AuthContainerMount } = initSetup();
  const signInTest = AuthContainerMount.props().signIn;
  const ButtonTest = AuthContainerMount.find('button');
  ButtonTest.simulate('click');
  expect(signInTest.mock.calls.length).toEqual(1);
});
