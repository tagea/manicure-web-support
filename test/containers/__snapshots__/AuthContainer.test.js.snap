// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render AuthContainer 1`] = `
<Auth
  changeInput={[Function]}
  error={false}
  handleSubmit={[Function]}
  pressEnter={[Function]}
/>
`;
